### PR TITLE
Fix the indentation produced by max-indent

### LINKED ISSUE #925: Bug: inconsistent indentation when using max-indent=2
Ocamlformat 0.10 configuration:

```
max-indent=2
break-separators=after-and-docked
```

The code is indented this way. You can see that the `2.;` starts on a different column than all the other elements. I think `2.` should be aligned like the other ones.

```
let array =
  [| 0.5;
    2.;
     3.5;
     5.;
     6.5;
     8.;
     9.5;
     11.;
     12.5;
     14.;
     15.5;
     17.;
     18.5;
     20.;
     21.5;
     23.;
     24.5;
     26.;
     27.5;
     29.;
     30.5;
     32.;
     33.5;
     35.;
     36.5;
     38.;
     39.5;
     41.;
     42.5;
     44.
  |]
```

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #925 (in progress)

### CODE REVIEW

File: test/passing/max_indent.ml [hunk 1]
```diff
@@ -7,8 +7,36 @@ let () =
...
+        (fun lbl -> (mknoloc (Longident.Lident "?temp?"), lbl, omega))
+        lbl_all
+    in
+    fooooooo
+
+let foooooooooo =
+  match fooooooooooooo with
+  | Pexp_construct
+    ({txt= Lident "::"; _}, Some {pexp_desc= Pexp_tuple [_; e2]; _}) ->
+    if is_sugared_list e2 then Some (Semi, Non)
```
Comment (gpetiot) [inline test/passing/max_indent.ml]: to be fixed

**Comment by Julow** [comment]
With `max-indent=1` I have this:

```ocaml
 match foooooooooooooooooooooo with
 | ( Ppat_constraint
  ( ({ppat_desc= Ppat_var _; _} as p0)
  , {ptyp_desc= Ptyp_poly ([], t0); _} )
   , Pexp_constraint (e0, t1) )
   when Poly.(t0 = t1) ->
    m.value_binding m
```

Shoudn't this be like this ?

```ocaml
 match foooooooooooooooooooooo with
 | ( Ppat_constraint
  ( ({ppat_desc= Ppat_var _; _} as p0)
  , {ptyp_desc= Ptyp_poly ([], t0); _} )
  , Pexp_constraint (e0, t1) )
  when Poly.(t0 = t1) ->
   m.value_binding m
```

**Comment by gpetiot** [comment]
(only half-joking) maybe we should only accept `--max-indent=i` with `i >= 2` ?

**Comment by Julow** [comment]
(the other half-joking) or we could implement it as the max indent a box can add and not on the absolute indentation across lines ?

For example, with `max-indent=1`, this could be formatted like this:

```ocaml
 match foooooooooooooooooooooo with
 | ( Ppat_constraint
   ( ({ppat_desc= Ppat_var _; _} as p0)
   , {ptyp_desc= Ptyp_poly ([], t0); _} )
  , Pexp_constraint (e0, t1) )
  when Poly.(t0 = t1) ->
   m.value_binding m
```

Even if the nested tuple is indented 2 spaces more than the line before, it is indented 1 space more than the whole pattern that is indented 1 space more than the whole match

**Comment by gpetiot** [comment]
Replaced by #1118

**Pull request closed without merging**